hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1490,6 +1509,13 @@ private void deleteInternal(Container container, boolean force)
         // 4. container moved to tmp folder
         // 5. container content deleted from tmp folder
         try {
+          long waitTime = clock.millis() - startTime;
+          if (waitTime > maxDeleteLockWaitMs) {
```

Comment:
How about to check the SCM command's deadline time too in here, We can wait longer if the SCM deadline time is not exceeded here.

Reply:
We can consider that as an improvement if this doesn't work well. As it stands, we are not sure if the long running commands were due to lock waits or slow deletes. It seems most likely it was slow deletes, but the logs had all rolled off on the cluster before we got to them. For now I think it makes sense to keep this simple.